### PR TITLE
Add warning on documentation for stop of python 2 support.

### PR DESCRIPTION
**Pull request by Carreau**

### CODE REVIEW

**Comment by Carreau** [comment]
@willingc I would appreciate a feedback on this one as it will adds this warning to all page of our documentation.

File: docs/source/conf.py [hunk 1]
```diff
@@ -93,13 +95,22 @@ def is_stable(extra):
     tags.add('ipystable')
 else:
     tags.add('ipydev')
-    rst_prolog = """
-    .. warning::
+    rst_prolog += """
+.. warning::
+
+    This documentation is for a development version of IPython. There may be
```
Comment (willingc) [inline docs/source/conf.py]: This documentation covers a development version of IPython. The development version may differ significantly from the latest stable release.

File: docs/source/conf.py [hunk 2]
```diff
@@ -93,13 +95,22 @@ def is_stable(extra):
...
+
+rst_prolog += """
+.. important::
 
-        This documentation is for a development version of IPython. There may be
-        significant differences from the latest stable release.
+    This is the documentation for IPython version > 6.0 which is had
+    stopped compatibility for python version lower than 3.3. If you are
+    looking for a version of IPython compatible with python 2.7 please see
+    the documentation for the IPython 5.x LTS (Long term support branch)
```
Comment (willingc) [inline docs/source/conf.py]: This documentation covers IPython versions 6.0 and higher. Beginning with version 6.0, IPython stopped supporting compatibility with Python versions lower than 3.3 including all versions of Python 2.7.

If you are looking for an IPython version compatible with Python 2.7, please use the IPython 5.x LTS release and refer to its documentation (LTS is the long term support release).

File: docs/source/coredev/release_process.rst [hunk 3]
```diff
@@ -170,6 +170,12 @@ Get a fresh clone of the tag for building the release::
 8. Run the release script
 -------------------------
 
+.. important::
+
+    Following releases instructions have information to release IPython 5.x and
```
Comment (willingc) [inline docs/source/coredev/release_process.rst]: These steps cover instructions for creating releases of IPython 5.x LTS and IPython 6.x. Ignore release steps for Python 2 when releasing IPython 6.x which no longer supports Python 2.

File: docs/source/overview.rst [hunk 4]
```diff
@@ -239,7 +239,8 @@ This functionality is optional and now part of the `ipyparallel
 Portability and Python requirements
 -----------------------------------
 
-As of the 2.0 release, IPython works with Python 2.7 and 3.3 or above.
+Version 6.0 of IPython work with python 3.3 and above.
```
Comment (willingc) [inline docs/source/overview.rst]: Version 6.0+ supports compatibility with Python 3.3 and higher.
Versions 2.0 to 5.x work with Python 2.7.x releases and Python 3.3 and higher.

**Comment by willingc** [comment]
@Carreau 🏄 🐙 🌴 Comments left :D

**Comment by Carreau** [comment]
I'm amazed by your knowledge in emoji. 

Updated with your comments !

**Pull request merged**